conda/recipes/libcuml/meta.yaml

```diff
@@ -67,21 +85,31 @@ outputs:
       number: {{ GIT_DESCRIBE_NUMBER }}
       string: cuda{{ cuda_major }}_{{ date_string }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
       ignore_run_exports_from:
-        - {{ compiler('cuda') }}
+        {% if cuda_major == "11" %}
+        - {{ compiler('cuda11') }}
+        {% endif %}
     requirements:
       build:
         - cmake {{ cmake_version }}
       run:
-        - cudatoolkit {{ cuda_spec }}
-        - libcublas {{ libcublas_run_version }}
-        - libcublas-dev {{ libcublas_run_version }}
-        - libcufft {{ libcufft_run_version }}
+        - {{ pin_compatible('cuda-version', max_pin='x', min_pin='x') }}
+        {% if cuda_major == "11" %}
+        - cudatoolkit
+        - libcublas {{ cuda11_libcublas_run_version }}
+        - libcublas-dev {{ cuda11_libcublas_run_version }}
+        - libcufft {{ cuda11_libcufft_run_version }}
+        - libcurand {{ cuda11_libcurand_run_version }}
+        - libcusparse {{ cuda11_libcusparse_run_version }}
+        - libcusparse-dev {{ cuda11_libcusparse_run_version }}
+        {% else %}
+        - cuda-cudart-dev {{ cuda_spec }}
+        - libcublas-dev {{ cuda11_libcublas_run_version }}
+        - libcufft {{ cuda11_libcufft_run_version }}
+        - libcurand {{ cuda11_libcurand_run_version }}
+        - libcusparse {{ cuda11_libcusparse_run_version }}
```

Comment:
Think we want to drop the CUDA 11 version constraints from the `else` (CUDA 12) case like so

Mentioning as there is [a CI error]( https://github.com/rapidsai/cuml/actions/runs/5513300519/jobs/10051937250?pr=5318#step:7:328 ) that looks related to this

<details>

```
Could not solve for environment specs
The following package could not be installed
└─ libcuml-tests   is uninstallable because it requires
   └─ libcuml 23.08.00a cuda12_230710_g4c498326f_58, which requires
      ├─ libcumlprims 23.08.*  with the potential options
      │  ├─ libcumlprims 23.08.00a would require
      │  │  └─ cuda-version >=11,<12.0a0 , which can be installed;
      │  ├─ libcumlprims 23.08.00a would require
      │  │  └─ cudatoolkit >=11,<12.0a0 , which can be installed;
      │  └─ libcumlprims 23.08.00a would require
      │     └─ libcurand >=10.3.1.50,<11.0a0 , which can be installed;
      └─ libcurand >=10.2.5.43,<=10.3.0.86 , which conflicts with any installable versions previously reported.
```

</details>

```suggestion
        - cuda-cudart-dev
        - libcublas-dev
        - libcufft
        - libcurand
        - libcusparse
```